subworkflows/local/generate_downstream_samplesheets/main.nf

```diff
@@ -0,0 +1,112 @@
+                                        def out_path            = file(params.outdir).toString() + '/filter/filtered/'
+                                        def sample              = meta.id
+                                        def run_accession       = meta.id - "_longReads"
+                                        def instrument_platform = !meta.long_reads ? "ILLUMINA" : "OXFORD_NANOPORE"
+                                        def fastq_1             = meta.single_end  ?  out_path + reads.getName(): out_path + reads[0].getName()
+                                        def fastq_2             = !meta.single_end ? out_path + reads[1].getName() : ""
+                                        def fasta               = ""
+                                    [ sample: sample, run_accession:run_accession, instrument_platform:instrument_platform, fastq_1:fastq_1, fastq_2:fastq_2, fasta:fasta ]
+                                }
+                                .tap{ ch_colnames } //ch_header exists using ch_colnames instead
+
+    channelToSamplesheet(ch_colnames, ch_list_for_samplesheet, 'downstream_samplesheets', 'taxprofiler', format, format_sep)
+
+}
+
+workflow SAMPLESHEET_MAG{
+    take:
+    ch_reads
+
+    main:
+    ch_header  = Channel.empty()
+    format     = 'csv' // most common format in nf-core
+    format_sep = ','
+
+    ch_reads
+        .map{meta, reads ->
+                tuple( groupKey(meta.id - "_longReads", 2), meta, reads)
+            }
+        .groupTuple(remainder: true)
+        .map{key, meta, reads ->
+            new_meta = [
+                id: key,
+                run: key,
+                single_end: meta[0].single_end,
+                long_reads: meta[0]?.long_reads ?: meta[1]?.long_reads ?: false
+                ]
+            // Making sure the long reads are the final element of the array.
+            read_files = reads.flatten().sort(false){ a, b -> a.getName().tokenize('.')[0] <=> b.getName().tokenize('.')[0] }
+            [new_meta, read_files]
+            }
+        .tap{ ch_reads_grouped }
+
+    // Make your samplesheet channel construct here depending on your downstream
+    ch_list_for_samplesheet = ch_reads_grouped
+                                .map {
+                                    meta, reads ->
+                                        def out_path       = file(params.outdir).toString() + '/filter/filtered/'
+                                        def sample         = meta.id
+                                        def run            = meta.run
+                                        def group          = ""                                                                                       // only used for co-abundance in binning
+                                        def short_reads_1  = meta.long_reads == (reads.size() > 2) ? out_path + reads[0].getName() : ""                // If long reads, but no short reads, then short_reads_1 is empty
```

Comment:
Is the comment referring to mag or detaxier?

Reply:
mag, it's just explainig the if statement